src/core/profiles/service/profiles.queries.ts

```diff
@@ -26,7 +26,19 @@ export const RANDOM_PROFILES_PAGINATE_QUERY = `
         FROM "profile_skills_skill" "profile_skills"
         JOIN "skill" "skills" ON "skills"."id"="profile_skills"."skillId"
         WHERE "profile_skills"."profileId" = "profile"."id"
-      ) AS "skills"
+      ) AS "skills",
+      ARRAY(
+        SELECT DISTINCT ON ("language_profile"."id") json_build_object(
+          'id', "language_profile"."id",
+          'profileId', "language_profile"."profileId",
+          'level', "language_profile"."level",
+          'name', "languages"."name",
+          'languageId', "language_profile"."languageId"
+        )::text
+        FROM "language_profile"
+        JOIN "language" "languages" ON "languages"."id"="language_profile"."languageId"
+        WHERE "language_profile"."profileId" = "profile"."id"
+      ) AS "languageProfile"
```

Comment:
Servira si hago esto?
```suggestion
      ARRAY(
        SELECT DISTINCT ON (lgp.id) json_build_object(
          id, lgp.id,
          profileId, lgp.profileId,
          level, lgp.level,
          name, lg.name,
          languageId, lgp.languageId
        )::text
        FROM language_profile lgp
        JOIN language lg ON "lg.id=lgp.languageId
        WHERE lgp.profileId = profile.id
      ) AS languageProfile
```

Reply:
Otra cosa, deberia ser languagues unicamente, no languagueProfile ya que si bien lo buscas en la pivote, el frontend sabe que esos son los lenguages del perfil ps... (amenos que le cambies el nombre luego en la capa de presentacion y no lo quieras cambiar aqui)

```suggestion
      ARRAY(
        SELECT DISTINCT ON ("language_profile"."id") json_build_object(
          'id', "language_profile"."id",
          'profileId', "language_profile"."profileId",
          'level', "language_profile"."level",
          'name', "languages"."name",
          'languageId', "language_profile"."languageId"
        )::text
        FROM "language_profile"
        JOIN "language" "languages" ON "languages"."id"="language_profile"."languageId"
        WHERE "language_profile"."profileId" = "profile"."id"
      ) AS languages // <------------------------------
```

Reply:
esta dudoso que eso funcione con el lgp.id xD, ya si cambiar el AS languages o cambiarlo en la presentacion de la respuesta ta bien